### ISSUE TITLE
Anomaly when using Polymorphic Inductive Cumulativity

### ISSUE THREAD

**Issue by beta-ziliani** [coq/coq#8288]
#### Version
8.8.0

#### Operating system
Ubuntu 18.04.1

#### Description of the problem

The code in the gist below raises an anomaly `Invalid subtyping relation`.  It is using `Set Polymorphic Inductive Cumulativity`. If I remove that line, it works. BTW, most likely we shouldn't use inductive cumulativity in this code.

https://gist.github.com/beta-ziliani/2779d3a9ede67fc911c78a24890598c2

**Comment by SkySkimmer** [comment]
Confirmed in master.

**Comment by SkySkimmer** [comment]
PR with fix and much smaller reproduction #8290 
~~~coq
Set Universe Polymorphism.
Set Printing Universes.

Set Polymorphic Inductive Cumulativity.

Inductive foo := C : (forall A : Type -> Type, A Type) -> foo.
(* anomaly invalid subtyping relation *)
~~~
